discord-scripts/invite-management.ts

```diff
@@ -189,49 +189,98 @@ export default async function sendInvite(discordClient: Client, robot: Robot) {
             return
           }
 
-          const internalChannel = await interaction.guild.channels.create({
-            name: internalChannelName,
-            type: ChannelType.GuildText,
-            parent: defenseCategory.id,
-          })
-
-          const externalChannel = await interaction.guild.channels.create({
-            name: externalChannelName,
-            type: ChannelType.GuildText,
-            parent: defenseCategory.id,
-          })
+          // Internal channel setup
+          let internalChannel = interaction.guild.channels.cache.find(
+            (channel) => channel.name === internalChannelName,
+          ) as TextChannel
+          const internalChannelCreated = !internalChannel
+          if (internalChannelCreated) {
+            internalChannel = await interaction.guild.channels.create({
+              name: internalChannelName,
+              type: ChannelType.GuildText,
+              parent: defenseCategory.id,
+            })
+          }
 
           const internalRoleName = `Defense Internal: ${clientName.value}`
-          const externalRoleName = `Defense External: ${clientName.value}`
-          const internalRole = await interaction.guild.roles.create({
-            name: internalRoleName,
-            reason: "Role for internal audit channel",
-          })
-          const externalRole = await interaction.guild.roles.create({
-            name: externalRoleName,
-            reason: "Role for external audit channel",
-          })
-
-          await internalChannel.permissionOverwrites.create(internalRole, {
-            ViewChannel: true,
-          })
-          await externalChannel.permissionOverwrites.create(externalRole, {
-            ViewChannel: true,
-          })
-          await externalChannel.permissionOverwrites.create(internalRole, {
-            ViewChannel: true,
-          })
+          let internalRole = interaction.guild.roles.cache.find(
+            (r) => r.name === internalRoleName,
+          )
+          if (!internalRole) {
+            internalRole = await interaction.guild.roles.create({
+              name: internalRoleName,
+              reason: "Role for internal audit channel",
+            })
+          }
 
+          if (internalChannel) {
+            await internalChannel.permissionOverwrites.create(internalRole, {
+              ViewChannel: true,
+            })
+            await internalChannel.send(
+              `@here **Welcome to the ${clientName.value} Internal Audit Channel!**`,
```

Comment:
Sorry, what I meant with this message is that we should keep the full original version. i.e.:

```
**Welcome to the ${clientName.value} audit internal channel!**

You can use this invite to ...: ${internalInvite.url}.
```

And similar for external except it would say “Welcome to the ${clientName.value} audit channel...” (note that it doesn't say “external”, it just doesn't say “internal”).

The goal here is to have a point (possibly pinned) in each channel where folks can:
- See the invite code.
- See information about the invite code (e.g. when it expires and max uses).

Basically the same stuff we were showing @ https://github.com/thesis/valkyrie/pull/301/files#diff-b34d961b5563321a08dc8493fd258c4f5cec44c90e8edd92999c529b43b0f7aaL87-L98 .

Reply:
That makes sense and will get this changed over! Only question I have about that, if we pin the invite information to each channel, should we worry about deleting that message once a client joins or just simply always have it visible, and just assume team members can remove message if needed. I love the expiry / uses idea, that makes a lot of sense and also helps track.

Another layer, we could even update pinned message with a `@user` of who has claimed invite? If it seems important to surface.

Reply:
Love the idea of updating users & counts, though I would say not top priority.